Added extensions for args when calling file_validate_extensions

IAMlKeno opened the pull request.

IAMlKeno committed:
Added extensions for args when calling file_validate_extensions

BrandonMorr commented:
Hard coding the extensions is an aggressive approach, and if a user uploads a file that the DSID isn't compatible with you get the error:
```
Error message
Only files with the following extensions are allowed: jpeg jpe jpg png gif. (for the set DSID)
```

Should probably build that list of acceptable file extensions dynamically.

IAMlKeno committed:
Added function to dynamically build allowable exts. list

IAMlKeno committed:
Update coding standards, remove unused foreach

IAMlKeno committed:
Update to display appropriate exts for DS replace form

IAMlKeno committed:
Replaced xls extension addition

qadan commented on islandora_plupload.module:
```diff
@@ -183,3 +189,30 @@ function islandora_plupload_file_save($file) {
+      $extensions .= " ";
+
+    }
+    // XXX Module disables acceptable file types without xls ext added.
```
This doesn't seem to be true; I don't see any difference in functionality with the next line commented out. It also breaks the convention we're establishing here that a datastream must have a mimetype associated from its DS composite model. Either way, there should be a better way of doing this.

qadan reviewed (changes requested):
I'm concerned that this doesn't belong in the core module, as it breaks one of the core conventions of Islandora, namely that in practice, an ingested datastream does not have strict extension/mime requirements based on DSID or anything else. The form affected doesn't have any validation whatsoever. If this is indeed a client requirement, it should go in the client module.

Additionally, if this is happening for one form, the solution presented here is a bit broad, as it affects every form under `/islandora/%islandora_object` ... other pages that don't currently have extension requirements would now get them. It should likely do a more precise check of the menu path if we want it to just be happening on the datastream upload/replace forms?

IAMlKeno committed:
Remove file type validation

- When adding new datastreams with no file type restrictions
- When replacing a datastream that accepts any file type

IAMlKeno committed:
Update coding standards

IAMlKeno committed:
Removed double dollar signs

qadan commented on islandora_plupload.module:
```diff
@@ -55,6 +55,33 @@ function islandora_plupload_alter_managed_files(&$form) {
+        // prevents users from adding datastreams and replacing datastreams that
+        // allow any file type as a result of validating a files extension
+        // (ex. .jpg) against an empty string.
+        $current_path = current_path();
```
Since that's just a wrapper around `$_GET['q']`, it's missing all that drupal-y hook/alter goodness and might break in some cases. Should use `menu_get_item()` instead, that would also allow you to match directly against `$menu_item['path'] == 'islandora/object/%/manage/datastreams/add'`

qadan commented on islandora_plupload.module:
```diff
@@ -55,6 +55,33 @@ function islandora_plupload_alter_managed_files(&$form) {
+            'islandora_plupload_element_processs',
+          );
+        }
+        elseif (preg_match('/\/datastream\/.*\/replace/', $current_path)) {
```
Same deal here, if using `menu_get_item()` could match directly against `$menu_item['path'] == 'islandora/object/%/datastream/%/replace'`. It'd also allow you to directly grab the datastream (and therefore DSID/parent) from looking at `$menu_item['page_arguments'][1]`

IAMlKeno committed:
Code feedback - updating best practices getting data from url

qadan commented on islandora_plupload.module:
```diff
@@ -55,6 +55,32 @@ function islandora_plupload_alter_managed_files(&$form) {
+        elseif ($menu_path['path'] === $replace_datastream_path) {
+          $dsid = $menu_path['page_arguments'][1]->id;
+          $object = $menu_path['page_arguments'][1]->parent;
+          if (isset($object)) {
```
One other small thing; `$object` will always be set here since it's part of the tuque object. Might be better to check that `$menu_path['page_arguments'][1]` isn't `FALSE`-y? `isset()` also can give false positives for object loading, since it can be `NULL` or `FALSE` depending on circumstances

IAMlKeno replied:
Updated to `if ($menu_path['page_arguments'][1]) {`

IAMlKeno committed:
Feedback - run condition check on page args
